### PR TITLE
[PROCS-4153] Create basic e2e k8s/kind test environment setup for the process-agent

### PR DESCRIPTION
**Pull request by wiyu**
### What does this PR do?
- Creates a new test skeleton with k8s/kind env setup for a process-agent e2e tests.
- Creates an initial helm template to control the dd-agent's configuration

### Motivation
PROCS-4153

### CODE REVIEW

**Review by daniel-taf — APPROVED** [review]

**Comment by wiyu** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Comment by wiyu** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Review by L3n41c — APPROVED** [review]

**Review by spencergilbert — APPROVED** [review]
Approved the BaRX owned file

File: test/new-e2e/tests/process/k8s_test.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	options := []e2e.SuiteOption{
+		e2e.WithProvisioner(awskubernetes.KindProvisioner(
+			awskubernetes.WithWorkloadApp(func(e config.Env, kubeProvider *kubernetes.Provider) (*kubeComp.Workload, error) {
+				return cpustress.K8sAppDefinition(e, kubeProvider, "workload-stress")
+			}),
+			awskubernetes.WithAgentOptions(kubernetesagentparams.WithHelmValues(helmValues)),
+		)),
+	}
+
+	devModeEnv, _ := os.LookupEnv("E2E_DEVMODE")
```
Comment (pducolin) [inline test/new-e2e/tests/process/k8s_test.go:75]: **💬 suggestion**
No need for this extra env variable, using `E2E_DEV_MODE` will automatically enable the dev mode in the test runner

**Review by pducolin — APPROVED** [review]

**Pull request merged**